web/src/components/SongQueueItem.js

```diff
@@ -2,6 +2,34 @@ import React from 'react';
 import classNames from 'classnames';
 import {timeStr} from '../util/song';
 
+const QueueItemStatus = {
+  INVALID: 0,
+  UNKNOWN: 1,
+  WAITING: 2,
+  DOWNLOADING: 3,
+  PROCESSING: 4,
+  DONE: 5,
+};
+
+function setStatus(status) {
+  switch (status) {
+    case QueueItemStatus.INVALID:
+      return 'Error';
+    case QueueItemStatus.UNKNOWN:
+      return 'adding...';
+    case QueueItemStatus.WAITING:
+      return 'waiting...';
+    case QueueItemStatus.DOWNLOADING:
+      return 'downloading...';
+    case QueueItemStatus.PROCESSING:
+      return 'processing...';
+    case QueueItemStatus.DONE:
+      return 'ready';
+    default:
+      return "You're not supposed to be here";

```

Comment:
Can we just make the default also be the `DONE` state? Which I think is *not* supposed to have a status string.